docs/types/dynamic-dispatch.md

```diff
@@ -162,3 +162,133 @@ main = Test.simplify
 When invoking a method on _module object_ its _module static methods_ take
 precedence over _instance methods_ defined on `Any`. Thus a module serves
 primarily as a _container for module (static) methods_.
+
+## Method Resolution
+
+**Terminology**:
+
+- **eigen type** of a type `My_Type` is a type of type, usually written as
+  `My_Type.type`.
+  - Every type has an eigen type.
+  - Eigen type of an eigen type is itself.
+- **associated type** of a module `My_Module` is a type for the module
+  - It is basically an eigen type for a module.
+- **symbol table**.
+  - Every type has an associated symbol table.
+  - Symbol table maps symbols to their definitions.
+  - All definitions are methods.
+    - Atom constructors are methods.
+    - Atom fields are methods. More specifically, every atom field has an
+      associated getter method.
+
+This section describes the _method resolution_ process, which resolves a
+concrete method definition for a concrete call site. For a method call
+expression `Receiver.symbol`, this section focuses only on a single dispatch
+based on `Receiver` argument. For multiple dispatch, see the
+[Multiple Dispatch](#multiple-dispatch) section.
+
+Method resolution algorithm for `Receiver.symbol` first determines the _type_ of
+the `Receiver`, and then finds the method definition in its _symbol table_:
+
+1. **Determine the type of `Receiver`:**
+
+- 1.1. If `Receiver` is type, the result will be _eigen type_
+- 1.2. If `Receiver` is a value (instance / atom), the result will be the _type
+  of the value_
+- 1.3. If `Receiver` is a module, the result will be the _associated type_ for
+  the module
+- 1.4. If `Receiver` is a polyglot object, method resolution and invocation will
+  be handled according to the [polyglot interoperability](../polyglot/README.md)
+  rules.
+  - Polyglot object can be:
+    - A Java class, imported by `polyglot java import ...` statement.
+    - Java object instance, created by `Java_Class.new ...` expression.
+    - Javascript, Python, or any other allowed foreign language object returned
+      by a foreign method call.
+- 1.5. If there is no `Receiver`, we are just looking for a variable in the
+  current lexical scope or any parent scopes. See
+  [Lexical scope lookup](#lexical-scope-lookup).
+
+2. **Look up symbol in the symbol table of the determined type:**
+
+- 2.1. If `symbol` is defined in the table, select it and stop.
+- 2.2. If parent type is present, repeat the process with the parent type.
+  - Every type has an implicit parent type `Any`.
```

Comment:
```suggestion
  - There is a special type called `Any` at the root of "class hierarchy"
  - Every type has an implicit parent type `Any`.
```

Reply:
Already explained in the terminology now.